src/SldStyleParser.ts

```diff
@@ -292,188 +284,203 @@ export class SldStyleParser implements StyleParser<string> {
-    if (!sldObject) {
-      return undefined;
-    }
-    const expressionOrValue =_get(sldObject, '$$[0]');
-    const expressionOrValueName = _get(sldObject, '$$[0]["#name"]');
-    if (expressionOrValueName === '__text__') {
-      return this.getValueFromSldObject(expressionOrValue);
-    } else {
-      return this.getExpressionFromSldObject(expressionOrValue);
-    }
+  /**
+   * Get the geostyler-style rules from a SLD Object (created with fast-xml-parser).
+   *
+   * @param sldObject The SLD object representation (created with fast-xml-parser)
+   * @return The geostyler-style rules
+   */
+  getRulesFromSldObject(sldObject: any): GsRule[] {
+    const layers = getChildren(sldObject[0].StyledLayerDescriptor, 'NamedLayer');
+    const rules: GsRule[] = [];
+    layers.forEach(({NamedLayer: layer}: any) => {
+      getChildren(layer, 'UserStyle').forEach(({UserStyle: userStyle}: any) => {
+        getChildren(userStyle, 'FeatureTypeStyle').forEach(({FeatureTypeStyle: featureTypeStyle}: any) => {
+          getChildren(featureTypeStyle, 'Rule').forEach(({Rule: sldRule}: any) => {
+            const filter: GsFilter | undefined = this.getFilterFromRule(sldRule);
+            const scaleDenominator: ScaleDenominator | undefined = this.getScaleDenominatorFromRule(sldRule);
+            const symbolizers: Symbolizer[] = this.getSymbolizersFromRule(sldRule);
+            const ruleTitle = get(sldRule, 'Title.#text');
+            const ruleName = get(sldRule, 'Name.#text');
+            const name = ruleTitle !== undefined
+              ? ruleTitle
+              : (ruleName !== undefined ? ruleName : '');
+            const rule: GsRule = <GsRule> {
+              name
+            };
+            if (filter) {
+              rule.filter = filter;
+            }
+            if (scaleDenominator) {
+              rule.scaleDenominator = scaleDenominator;
+            }
+            if (symbolizers) {
+              rule.symbolizers = symbolizers;
+            }
+            rules.push(rule);
+          });
+        });
+      });
+    });
+    return rules;
   }
 
   /**
    * Get the name for the Style from the SLD Object. Returns the Title of the UserStyle
    * if defined or the Name of the NamedLayer if defined or an empty string.
    *
-   * @param {object} sldObject The SLD object representation (created with xml2js)
-   * @return {string} The name to be used for the GeoStyler Style Style
+   * @param sldObject The SLD object representation (created with fast-xml-parser)
+   * @return The name to be used for the GeoStyler Style Style
    */
   getStyleNameFromSldObject(sldObject: any): string {
-    const userStyleTitle = _get(sldObject, 'StyledLayerDescriptor.NamedLayer[0].UserStyle[0].Title[0]._');
-    const namedLayerName = _get(sldObject, 'StyledLayerDescriptor.NamedLayer[0].Name[0]._');
+    const userStyleTitle = get(sldObject, 'StyledLayerDescriptor.NamedLayer[0].UserStyle.Title.#text');
```

Comment:
Someday we will properly distinguish between style name and style title. Could you add an issue for that?

Reply:
https://github.com/geostyler/geostyler-sld-parser/issues/new/choose